modules/core/src/main/java/org/locationtech/jts/geom/ZInterpolating.java

```diff
@@ -0,0 +1,40 @@
+package org.locationtech.jts.geom;
+
+/**
+ * Internal class which encapsulates the runtime switch to Z-interpolate when applicable.
+ * <p>
+ * <ul>
+ * <li><code>jts.zinterpolating=false</code> - (default) do not Z-interpolate
+ * <li><code>jts.zinterpolating=true</code> - Z-interpolate
+ * </ul>
+ * 
+ * @author bjornharrtell
+ *
+ */
+public class ZInterpolating {
```

Comment:
This seems a lot of boilerplate to hold a boolean value decision, can this class take on more responsibility and swap between interpolation approaches as a stratagy object?

Reply:
I agree it feel like boilerplate. I have difficulties properly understanding/interpreting what you mean with "take on more responsibility and swap between interpolation approaches as a stratagy object". In either case it's intentionally done this way because it is modeled after the other existing system property option jts.overlay and broken out into a self contained class because it is affecting two very separate things (RobustLineIntersector and ElevationModel), and I wanted to do it in a way that is as non intrusive as possible on the existing source.

I suppose system wide settings could be done with smarter and nicer desig but I don't think this is the place to do such general improvement and restructuring.